### DISCUSSION TITLE
Episode 226, concurrency warning

### DISCUSSION THREAD

**Discussion by sbeitzel** [pointfreeco/episode-code-samples#136] [Q&A] [answered]
So, in the `ifLet` function defined for Reducer, we've got this code:

```swift
                    onFirstAppearEffect = .run { send in
                        do {
                            try await withTaskCancellation(id: DismissID(id: childStateAfter.id)) {
                                try await Task.never()
                            }
                        } catch is CancellationError {
                            await send(actionCasePath.embed(.dismiss))
                        }
                    }
                    .cancellable(id: childStateAfter.id)
```

I've got concurrency warnings cranked way up and as a result, Xcode is complaining about the `withTaskCancellation` line and the `await send...` line. First, it says: "Capture of 'childStateAfter' with non-sendable type 'ChildState' in a `@Sendable` closure" and then it says, "Non-sendable type 'Self.Action' passed in call to main actor-isolated function cannot cross actor boundary"

I guess the question I have is, is this a problem?

**Comment by stephencelis** [comment, marked as answer]
@sbeitzel The Composable Architecture was built before Swift concurrency and has not been audited for strict concurrency yet. As helpful as Swift's concurrency diagnostics are when they work, we've found them to be a little flakey and buggy, and so we've been waiting for the tools to mature before fully adopting them.

In this case the warnings can generally be fixed by adding some `Sendable` conformances. We'll try to clean up some of this in time for the navigation release, though, with the help of PRs like https://github.com/pointfreeco/swift-composable-architecture/pull/1994.
